src/lib/containers/VisibilityObserver.tsx

```diff
@@ -82,25 +82,29 @@ export default class VisibilityObserver extends React.Component<
     const { rootMargin, tag } = this.props
 
     return (
-      <Observer
+      <InView
         as={(tag as any) || 'div'}
         rootMargin={rootMargin}
         onChange={this.onVisibilityChange}
       >
         {this.getObserverChildren}
-      </Observer>
+      </InView>
     )
   }
 
-  private getObserverChildren = (isVisible: boolean) => {
+  private getObserverChildren = (renderProps: {
+    inView: boolean
+    entry: IntersectionObserverEntry | undefined
+    ref: React.RefObject<any> | ((node?: Element | null) => void)
+  }) => {
     const { renderInView, renderOutOfView } = this.props
-    const shouldRenderAsInView = isVisible || this.state.hasBeenInView
+    const shouldRenderAsInView = renderProps.inView || this.state.hasBeenInView
 
     if (shouldRenderAsInView && renderInView) {
       return renderInView()
     }
 
-    if (!isVisible && renderOutOfView) {
+    if (!renderProps.inView && renderOutOfView) {
       return renderOutOfView()
     }
```

Comment:
I was getting type errors before the 'react-intersection-observer' upgrade, so I wonder how well this component was actually working.